### PR TITLE
CAMEL-16542 Camel-Solr: Refactoring the component a bit

### PR DESCRIPTION
**Pull request by Paul-Blanchaert**

### CODE REVIEW

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java [hunk 1]
```diff
@@ -86,38 +113,105 @@ public SolrComponent() {
...
     @Override
     protected void doShutdown() throws Exception {
+        removeFromSolrClientMap(solrClientMap.keySet());
         for (SolrServerReference server : servers.values()) {
             shutdownServers(server);
         }
         servers.clear();
     }
 
+    @Deprecated
```
Comment (oscerd) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java:201]: If those methods are not used anymore, you can safely remove them. We are going to release an LTS release with 3.11.0, so we are under heavy developments, so you can remove methods like these.

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+            }
+        }
+        return urlList;
+    }
+
+    /**
+     * signature defines parameters deciding whether or not to share the solrClient - sharing allowed: same signature -
+     * sharing not allowed: different signature
+     */
+    public String getSignature() {
```
Comment (oscerd) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java:366]: These seems utility methods. I think they should be in a separated class instead that the configuration class.

**Review by oscerd — CHANGES_REQUESTED** [review]
Just some comments, but looks really promising. Thanks for this work! @orpiske can you have a look at the test-infra part please? Thanks a lot @Paul-Blanchaert !

**Comment by orpiske** [comment]
> Just some comments, but looks really promising. Thanks for this work! @orpiske can you have a look at the test-infra part please? Thanks a lot @Paul-Blanchaert !

Sure thing! I take a look tomorrow. Thanks for the contribution @Paul-Blanchaert

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java [hunk 3]
```diff
@@ -17,8 +17,7 @@
 package org.apache.camel.component.solr;
 
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (omarsmak) [inline components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java:20]: Please remove the wildcard import

**Review by orpiske — APPROVED** [review]
Test infra LGTM

**Comment by oscerd** [comment]
@Paul-Blanchaert feel free to update the PR when you have time and fix the findings (not so much), thank you!

File: test-infra/camel-test-infra-solr/pom.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>camel-test-infra-parent</artifactId>
+        <groupId>org.apache.camel</groupId>
+        <relativePath>../camel-test-infra-parent/pom.xml</relativePath>
+        <version>3.10.0-SNAPSHOT</version>
```
Comment (oscerd) [inline test-infra/camel-test-infra-solr/pom.xml:27]: We released 3.10.0, so this needs to be moved to 3.11.0-SNAPSHOT and the PR needs to be aligned. Thanks!

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java (hunk 1 above)
Comment (Paul-Blanchaert) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java]: ok, thanks

File: components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java (hunk 3 above)
Comment (Paul-Blanchaert) [reply components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrComponent.java]: ok, done

File: test-infra/camel-test-infra-solr/pom.xml (hunk 4 above)
Comment (Paul-Blanchaert) [reply test-infra/camel-test-infra-solr/pom.xml]: ok, done

**Comment by oscerd** [comment]
There is a compilation failure, just delete the src/generated folder, it will be regenerated if you rebuild and the build should be fine.

**Comment by davsclaus** [comment]
Is this PR ready or does it need more work?

**Comment by Paul-Blanchaert** [comment]
I was planning to put some more work in (esp. tests) somewhere next week or
week after. Is there a rush or deadline we should consider?

**Comment by oscerd** [comment]
This can be merged. We can add more tests with other PRs

**Comment by Paul-Blanchaert** [comment]
Fine with me...
I promise to Andrea that I'll deal with his last open comment on this PR in
the next one ;-)

**Pull request merged**